src/facade/dragonfly_connection.h

```diff
@@ -79,6 +79,11 @@ class Connection : public util::Connection {
 
   struct MonitorMessage : public std::string {};
 
+  struct AclUpdateMessage {
+    std::string_view username;
+    uint64_t categories{0};
+  };
```

Comment:
Please add a comment above why we have it at all

Reply:
And quick check: are you sure the string_view keeps valid forever? The connection might be very busy and process this message only after a long delay

Reply:
Yes I am sure, because we hold a `ReadLock` on the registry so nobody can actually modify it:

```
180     auto& registry = ServerState::tlocal()->user_registry;                                           
181     auto user_with_lock = registry->MaybeAddAndUpdateWithLock(username, std::move(req));             
182     if (user_with_lock.exists) {                                                                     
183       StreamUpdatesToAllProactorConnections(username, user_with_lock.user.AclCategory());            
184     }  
```

Reply:
which is a bug! I will fix it, we need a `unique_lock`!